packages/bsky/src/services/actor/index.ts

```diff
@@ -77,33 +83,57 @@ export class ActorService {
     })
   }
 
-  searchQb(searchField: 'did' | 'handle' = 'handle', term?: string) {
+  async getSearchResults({
+    cursor,
+    limit = 25,
+    term = '',
+  }: {
+    cursor?: string
+    limit?: number
+    term?: string
+  }) {
+    const searchField = term.startsWith('did:') ? 'did' : 'handle'
+    let builder
     const { ref } = this.db.db.dynamic
-    let builder = this.db.db.selectFrom('actor')
-
-    // When searchField === 'did', the term will always be a valid string because
-    // searchField is set to 'did' after checking that the term is a valid did
-    if (searchField === 'did' && term) {
-      return builder.where('actor.did', '=', term)
+    const paginationOptions = {
+      limit,
+      cursor,
+      direction: 'asc' as const,
     }
+    let keyset
 
-    if (term) {
-      builder = builder.where((qb) => {
-        // Performing matching by word using "strict word similarity" operator.
-        // The more characters the user gives us, the more we can ratchet down
-        // the distance threshold for matching.
-        const threshold = term.length < 3 ? 0.9 : 0.8
-        return qb
-          .where(distance(term, ref('handle')), '<', threshold)
-          .orWhereExists((q) =>
-            q
-              .selectFrom('profile')
-              .whereRef('profile.creator', '=', 'actor.did')
-              .where(distance(term, ref('displayName')), '<', threshold),
-          )
-      })
+    if (term && searchField === 'handle') {
+      keyset = new SearchKeyset(ref('distance'), ref('actor.did'))
+      // Matching user accounts based on handle
+      const accountsQb = getMatchingAccountsQb(this.db, { term })
+        .orderBy('distance', 'asc')
+        .limit(limit)
+      // Matching profiles based on display name
+      const profilesQb = getMatchingProfilesQb(this.db, { term })
+        .orderBy('distance', 'asc')
+        .limit(limit)
+      // Combine and paginate result set
+      builder = combineAccountsAndProfilesQb(
+        this.db,
+        accountsQb,
+        profilesQb,
+      ).select('distance')
```

Comment:
I think this will have a pagination issue: since `accountQb` and `profilesQb` apply a limit but not any kind of offset, we wont be able to get deeper pages of results.  If you check out the `getUserSearchQuery()` utility, you can see how we handled this in `app.bsky.actors.searchActors`.  I wonder if we may even be able to use `getUserSearchQuery()` directly here, though!

Reply:
I see what you mean.
Adjusted the implementation to re-use `getUserSearchQuery` and a test to assert that pagination works.